### PR TITLE
Fixes for compatibility with pythia8 212

### PR DESCRIPTION
**Pull request by mkirsano**

### CODE REVIEW

**Comment by mkirsano** [comment]
Fixes for compatibility with pythia8 212 (PR cmsdist #1840). Compatible also with pythia8 210, so can be made first.

File: GeneratorInterface/Pythia8Interface/plugins/EmissionVetoHook1.h [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #include "Pythia8/Pythia.h"
 
+using namespace Pythia8;
```
Comment (bendavid) [inline GeneratorInterface/Pythia8Interface/plugins/EmissionVetoHook1.h]: Don't think we should have "using namespace" in a header file.

**Comment by bendavid** [comment]
The fix implemented here can cause a memory leak I think.  The pointer to fEvAttributes should be initialized to 0 in the constructor of LHAupLesHouches, then 
"fEvAttributes = new std::map<std::string, std::string >;"
could be replaced with
"if (fEvAttributes) {
  fEvAttributes->clear();
} else {
  fEvAttributes = new std::map<std::string, std::string >;
}
"

(then if the pointer is non-zero it should also be deleted in the destructor for LHAupLesHouches)

**Comment by mkirsano** [comment]
Thanks, Josh, I correct it. I always forget that there can be several runs.

**Comment by bendavid** [comment]
+1
(contingent on taking https://github.com/cms-sw/cmsdist/pull/1840 of course)

**Comment by bendavid** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**